### PR TITLE
Added a c++ library for working with fractions.

### PR DESCRIPTION
**Pull request by Niraj-Kamdar**
Congrats on taking the first step to contributing to the Sample Programs repository maintained by [The Renegade Coder][1]! For simplicity, please make sure that your pull request includes one and only one contribution.

## Complete the Applicable Sections Below

Find which section best describes your pull request and make sure you fill it out. To start, let us know which issue you've fixed.

- [ ] I fixed #your-issue-number-here

### Code Snippets

- [x] I named the pull request using `Added/Updated <Sample Program> in <Language>` format
- [x] I created/updated the language README
  - [x] I added the sample program name to the README
  - [x] I added fun facts (i.e. debut, developer, typing, etc.)
  - [ ] I added reference link(s) to the README
  - [ ] I added solution citations when necessary (see [plagiarism][2])

### Documentation

- [ ] I named the pull request using `Added/Updated <Sample Program> in <Language> Article` format
- [x] I followed the applicable article template
  - [ ] I added/updated a [language article][4] (i.e. The Python Programming Language)
  - [x] I added/updated a [code snippet article][3] (i.e. Hello World in Perl)
  - [x] I added/updated a [project article][5] (i.e. Fizz Buzz in Every Language)

### Testing

- [ ] I named the pull request using `Added/Updated <Language>/<Project> Testing` format
- [ ] I followed the testinfo template, if applicable

## Notes

Feel free to describe what you added or updated.

[1]: https://therenegadecoder.com/
[2]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/.github/CONTRIBUTING.md#plagiarism
[3]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/CODE_ARTICLE_TEMPLATE.md
[4]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/LANGUAGE_ARTICLE_TEMPLATE.md
[5]: https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/templates/PROJECT_ARTICLE_TEMPLATE.md

### CODE REVIEW

**Comment by jrg94** [comment]
This solution is a bit complicated, so I'll take a look shortly.

File: docs/projects/fractions/_posts/2019-10-10-c-plus-plus.md [hunk 1]
```diff
@@ -0,0 +1,251 @@
+---
+title: fractions in C++
+layout: default
+last-modified: 2019-10-10
+featured-image:
+tags: [c-plus-plus, fractions]
+authors:
+  - niraj-kamdar
```
Comment (jrg94) [inline docs/projects/fractions/_posts/2019-10-10-c-plus-plus.md]: Make sure you add yourself to the author file. https://github.com/TheRenegadeCoder/sample-programs/blob/master/docs/_data/authors.yml

File: docs/projects/fractions/_posts/2019-10-10-c-plus-plus.md [hunk 2]
```diff
@@ -0,0 +1,251 @@
...
+}
+```
+
+Here, we defined the `Fraction` class which we'll use to provide new data type for working with fractions.
+Inside this class, we begin by creating two private attributes numerator and denomenator.
+Constructor of fractions takes two optional arguments top and bottom where top will be assigned to numerator and
+bottom will be assigned to denometor.Here we checks if denometor is zeros then we throws an error and if denomenator is
+negative then we negate top and bottom before assigning them.
+We also have to implement getters for this Class and a method for unary operator `-`.
+We also have to implement friend functions(one can use normal method) for binary operators like +, -, >=, == etc.
```
Comment (jrg94) [inline docs/projects/fractions/_posts/2019-10-10-c-plus-plus.md]: Here, we try to include a section for how to run the solution. It appears your solution is more of a library and less of an executable program.

File: docs/projects/fractions/index.md [hunk 3]
```diff
@@ -0,0 +1,31 @@
...
+---
+
+languages like python have built-in utilities or functions for working with fractions.
+Many of these fractions functions follow a similar pattern across programming languages:
+Takes a numerator and a denomenator as an attribute.
+Perform basic arithmatic and relational operations with operator overloading.
+
+## Requirements
+
+In general, a fractions library should perform the following:
```
Comment (jrg94) [inline docs/projects/fractions/index.md]: Can we include some requirements that make this program executable? That way we could actually include some arithmetic or something on the command line.

**Comment by jrg94** [comment]
I just noticed the main function! Any chance we could standardize it for user input?

File: docs/projects/fractions/index.md (hunk 3 above)
Comment (Niraj-Kamdar) [reply docs/projects/fractions/index.md]: yep, i can redefine main such that it takes 3 external args ./fractions(compiled prog) operand1 operator operand2 and produces output.
EX: ./fractions "3/4" + "3/15" produces minimal p/q fraction 
output: 19/20
this way we can make it executable and easily usable fraction calculator program.

**Comment by Niraj-Kamdar** [comment]
I have changed main function as required.

**Comment by Niraj-Kamdar** [comment]
I don't know how to add article using syntax like this {% include article_list.md collection=site.categories.file-io %}.
so i am appending line -fill as needed to the article.

**Comment by jrg94** [comment]
Let me know when you think it's ready for review.

**Comment by Niraj-Kamdar** [comment]
You can review it now. I guess i have completed all the requirements.

File: docs/projects/fractions/_posts/2019-10-10-c-plus-plus.md (hunk 1 above)
Comment (jrg94) [reply docs/projects/fractions/_posts/2019-10-10-c-plus-plus.md]: I'm going to mention this again.

File: docs/projects/fractions/_posts/2019-10-10-c-plus-plus.md [hunk 4]
```diff
@@ -0,0 +1,702 @@
...
+    int getnumerator(){
+        return numerator;
+    }
+
+    int getdenometor(){
+        return denomenator;
+    }
+```
+
+# Friend functions for Fraction class
```
Comment (jrg94) [inline docs/projects/fractions/_posts/2019-10-10-c-plus-plus.md]:
```suggestion
### Friend functions for Fraction class 
```

File: docs/projects/fractions/index.md [hunk 5]
```diff
@@ -0,0 +1,32 @@
...
+1. Perform arithmatic operation like multiplications, addition etc.
+2. Give output for relational operations like >=, >, == etc.
+
+More specifically, begin with creating object instance of fraction class with two attributes:
+numerator and denomenator.Using operator overloading feature of langauge implement basic arithmatic
+and relational operaions.
+
+## Testing
+
+Verify that the actual output matches the expected output.
```
Comment (jrg94) [inline docs/projects/fractions/index.md]: Any chance you could put together a table of tests for your program?

File: docs/projects/fractions/index.md [hunk 6]
```diff
@@ -0,0 +1,32 @@
...
+
+- Fill as needed
```
Comment (jrg94) [inline docs/projects/fractions/index.md]: I'll fill in this part (when I remember)! 😄

Comment (Niraj-Kamdar) [reply docs/projects/fractions/index.md]: thanks

File: docs/projects/fractions/_posts/2019-10-10-c-plus-plus.md (hunk 1 above)
Comment (jrg94) [reply docs/projects/fractions/_posts/2019-10-10-c-plus-plus.md]: If you add yourself to the authors file, we'll be good to go.

**Comment by Niraj-Kamdar** [comment]
i have updated authors.yml

**Pull request merged**